test-loop-tests/src/utils/resharding.rs

```diff
@@ -852,3 +854,203 @@ pub(crate) fn check_state_cleanup(
     );
     LoopAction::new(action_fn, succeeded)
 }
+
+/// Repro case for the issue of 'Missing TrieValue' after GC period for refcounted trie nodes
+/// that are duplicated to both children during resharding. This particular scenario tests
+/// promise yield indices.
+pub(crate) fn promise_yield_repro_missing_trie_value(
+    left_child_account: AccountId,
+    right_child_account: AccountId,
+    gc_num_epochs: u64,
+    epoch_length: u64,
+) -> LoopAction {
+    let resharding_height: Cell<Option<u64>> = Cell::new(None);
+    let txs = Cell::new(vec![]);
+    let latest_height = Cell::new(0);
+    let nonce = Cell::new(102);
+    let yield_payload = vec![];
+    let pre_resharding_tx_sent = Cell::new(false);
+    let (checked_transactions, succeeded) = LoopAction::shared_success_flag();
+
+    let action_fn = Box::new(
+        move |node_datas: &[TestData],
+              test_loop_data: &mut TestLoopData,
+              client_account_id: AccountId| {
+            let client_actor =
+                retrieve_client_actor(node_datas, test_loop_data, &client_account_id);
+            let tip = client_actor.client.chain.head().unwrap();
+
+            // Function to send a promise yield receipt.
+            let send_promise_yield =
+                |signer_account: &AccountId,
+                 receiver_account: &AccountId,
+                 tip: &Tip,
+                 sent_flag: Option<&Cell<bool>>| {
+                    if sent_flag.map(|flag| flag.get()).unwrap_or_default() {
```

Comment:
nit: maybe `sent_flag.map_or(false, |flag| flag.get())` is slightly more readable

Reply:
Done